renku_notebooks/api/classes/data_service.py

```diff
@@ -246,3 +248,77 @@ def get_default_class(self) -> Dict[str, Any]:
 
     def find_acceptable_class(self, *args, **kwargs) -> Optional[ServerOptions]:
         return self.options
+
+
+@dataclass
+class GitProviderHelper:
+    """Calls to the data service to configure git providers."""
+
+    service_url: str
+    renku_url: str
+    internal_gitlab_url: str
+
+    def __post_init__(self):
+        self.service_url = self.service_url.rstrip("/")
+        self.renku_url = self.renku_url.rstrip("/")
+
+    def get_providers(self, user: User) -> list[GitProvider]:
+        connections = self.get_oauth2_connections(user=user)
+        providers: dict[str, GitProvider] = dict()
+        for c in connections:
+            if c.provider_id in providers:
+                continue
+            provider = self.get_oauth2_provider(c.provider_id)
+            access_token_url = urljoin(
+                self.renku_url,
+                urlparse(self.service_url + f"/oauth2/connections/{c.id}/token").path,
```

Comment:
it's already an f-string, just use `{self.service_url}` in the beginning instead of using `+`

Reply:
Done in d29ed41420c6b89d0ccbb60911289e2659412cd6.